lib/Runtime/Types/Type.h

```diff
@@ -62,6 +63,14 @@ namespace Js
         inline BOOL SkipsPrototype() const { return (this->flags & TypeFlagMask_SkipsPrototype) != 0 ; }
         inline BOOL CanHaveInterceptors() const { return (this->flags & TypeFlagMask_CanHaveInterceptors) != 0; }
         inline BOOL IsFalsy() const { return flags & TypeFlagMask_IsFalsy; }
+        inline BOOL HasBeenCached() const { return flags & TypeFlagMask_HasBeenCached; }
+        inline void SetHasBeenCached(const bool hasBeenCached)
+        {
+            // Once set, this flag should never be reset.
```

Comment:
> Do you mean caches are invalidated?

Yes, I meant inline caches holding those types get invalidated.

Yeah, I see what you mean. Probably having refcounting would help where every time you cache a type, you increase its refcount and we remove it, you decrease its refcount. If refcount == 0 means type is not cached. I am not sure how much benefit it will give but in big benchmarks like acmeair, lot of invalidation do happen and technically for those types, it is okay not to create new type for delete property.

---
In reply to: [78863703](https://github.com/Microsoft/ChakraCore/pull/1542#discussion_r78863703) [](ancestors = 78863703,78846360,78828446,78818005,78816821,78790053)